src/java.base/share/classes/java/util/Locale.java

```diff
@@ -398,76 +463,50 @@
- *     NumberFormat.getPercentInstance(myLocale);
- * }
- * A {@code Locale} is the mechanism for identifying the kind of object
- * ({@code NumberFormat}) that you would like to get. The locale is
- * <STRONG>just</STRONG> a mechanism for identifying objects,
- * <STRONG>not</STRONG> a container for the objects themselves.
- *
- * <h3>Compatibility</h3>
- *
- * <p>In order to maintain compatibility, Locale's
- * constructors retain their behavior prior to the Java Runtime
- * Environment version 1.7.  The same is largely true for the
- * {@code toString} method. Thus Locale objects can continue to
- * be used as they were. In particular, clients who parse the output
- * of toString into language, country, and variant fields can continue
- * to do so (although this is strongly discouraged), although the
- * variant field will have additional information in it if script or
- * extensions are present.
+ * <p>During deserialization, readResolve adds extensions as described
+ * in {@linkplain ##special_cases_constructor Special Cases}, only
+ * for the two cases th_TH_TH and ja_JP_JP.
+ *
+ * <h2>Compatibility</h2>
+ * @implNote
+ * <p> The following commentary is provided for apps that want to ensure
+ * interoperability with older releases of {@code Locale} provided by the
+ * reference implementation.
+ * <h3><a id="locale_behavior">Locale Behavior</a></h3>
+ * In order to maintain compatibility, Locale's (deprecated) constructors,
+ * {@link #of(String, String, String)}, and its overloads retain their behavior prior to the Java Runtime
+ * Environment version 1.7. That is, a length constraint is not imposed on any of
+ * the input parameters. Similarly, the same preservation of past behavior is largely true
+ * for the {@link #toString()} method.
+ * Apps that previously parsed the output of {@link #toString()} into language,
+ * country, and variant fields can continue to do so (although this is strongly
+ * discouraged). A caveat is that the variant field will have additional
+ * information in it if script or extensions are present.
  *
  * <p>In addition, BCP 47 imposes syntax restrictions that are not
  * imposed by Locale's constructors. This means that conversions
  * between some Locales and BCP 47 language tags cannot be made without
- * losing information. Thus {@code toLanguageTag} cannot
+ * losing information. Thus {@link #toLanguageTag} cannot
  * represent the state of locales whose language, country, or variant
  * do not conform to BCP 47.
  *
- * <p>Because of these issues, it is recommended that clients migrate
+ * <p>Because of these issues, it is recommended that apps migrate
  * away from constructing non-conforming locales and use the
- * {@code forLanguageTag} and {@code Locale.Builder} APIs instead.
- * Clients desiring a string representation of the complete locale can
- * then always rely on {@code toLanguageTag} for this purpose.
+ * {@link #forLanguageTag} and {@link Locale.Builder} APIs instead.
```

Comment:
Same remarks for links, add a override render text to prevent javadoc from generating long parameter lists.

Reply:
I gave the `Locale.of` links an explicit render. I think the `forLanguageTag` links were fine w/o an explicit render, as long as `{@link #forLanguageTag(String)}` was used, not `{@link #forLanguageTag}` (so that the fully qualified class name is omitted).

On a side note, the default render was pretty nasty for the `Locale.LanguageRange` see tags.